components/WorkSectionAsGallery.js

```diff
@@ -0,0 +1,221 @@
+  } else {
+    video.play();
+    document.getElementById(`${videoId}--video-button-text`).textContent = '(Pause)';
+  }
+}
+
+const renderWorkOverlay = (work) => {
+  const workIsImage = work.fields.asset.fields.file.contentType.startsWith("image/");
+
+  return (
+    <div className="WorkSectionAsGallery__work-hover-overlay">
+      <div className="WorkSectionAsGallery__work-hover-overlay--info">
+        <div className="WorkSectionAsGallery__work-hover-overlay--title">{work.fields.title}</div>
+        {work.fields.caseStudySlug ? 
+          <Link
+            href={work.fields.caseStudySlug}
+            passHref
+          >
+            <a
+              className="decoration-none color-white"
+              aria-label="read the case study"
+              rel="noopener noreferrer"
+            >
+            (read case study)
+            </a>
+          </Link> : 
+          <Link
+            href={work.fields.link}
+          >
+            <a
+              className="decoration-none color-white"
+              aria-label="visit the site"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+            → visit the site 
+            </a>
+          </Link>
+        }
+      </div>
+      {!workIsImage &&
+        <button 
+          id={`${work.sys.id}--video-button-text`}
+          className="WorkSectionAsGallery__work-hover-overlay--video-button"
+          onClick={() => pauseStartVideo(work.sys.id)}
+        >
+          (Pause)
+        </button>
+      }
+    </div>
+  );
+}
+
+const renderWorkAsImageOrVideo = (work) => {
+  const workIsImage = work.fields.asset.fields.file.contentType.startsWith("image/");
+  const workImage = flattenImageData(work.fields.asset);
+
+  return (
+    <div>
+      {workIsImage ? 
+        <Image
+          src={workImage.url}
+          width={workImage.width}
+          height={workImage.height}
+          sizes='50vw'
+        />  :
+        <video
+          id={work.sys.id}
+          key={work.sys.id}
+          style={{
+            width: '100%', 
+            height: '100%' 
+          }}
+          autoPlay
+          loop
+          muted
+          playsInline
+        >
+          <source src={work.fields.video.fields.file.url}></source>
+        </video>
+      }
+    </div>
+  );
+}
+
+const renderWorkGalleryBlock = (workGalleryBlock, index) => {
+  const fullWidthWork = get(workGalleryBlock, 'fields.fullWidthWork', {});
+  const halfWidthWorkOne = get(workGalleryBlock, 'fields.halfWidthWorkOne', {});
+  const halfWidthWorkTwo = get(workGalleryBlock, 'fields.halfWidthWorkTwo', {});
+
+  return (
+    <div className="flex-col">
+      {workGalleryBlock.fields.displayFullWidthAssetAtTop && 
+        <div className="block col-8 relative">
+          {renderWorkOverlay(fullWidthWork)}
+          {renderWorkAsImageOrVideo(fullWidthWork)}
+        </div>
+      }
+      <div className="flex col-8 pb_5 md:pb1 items-end" key={index}>
```

Comment:
don't think we need a key here for react. also we should not use[ indexes as keys](https://reactjs.org/docs/lists-and-keys.html). usually we can do something like `${work.id}-${index}`